src/mongo/db/commands/feature_compatibility_version.cpp

```diff
@@ -207,8 +207,6 @@ void FeatureCompatibilityVersion::set(OperationContext* txn, StringData version)
         // featureCompatibilityVersion to 3.4. This is a new index version that isn't supported by
         // versions of MongoDB earlier than 3.4 that will cause 3.2 secondaries to crash when it is
         // replicated.
-        std::vector<BSONObj> indexSpecs{k32IncompatibleIndexSpec};
```

Comment:
> Second, I want to make sure that the unused variable is just an accident, and doesn't signal that the code was incorrectly written or modified. Perhaps the variable is supposed to be being used!

The `indexSpecs` variable is indeed unnecessary. The initial version of the patch was modeled off logic in `CmdCreateIndex::run()` and used `MultiIndexBlock::init(const std::vector<BSONObj>& specs)` to create the index. While iterating on the patch, I realized that I could write a more compact of this logic by using the `IndexBuilder` class. It appears I mistakenly forgot to remove the `indexSpecs` variable when making that change.

> First, I would have expected the compiler to issue a warning, and I'd like to understand why it didn't in this case.

Are there particular `scons` options needed to cause such a warning to be emitted?